lib/file-source/src/paths_provider/glob.rs

```diff
@@ -14,54 +41,91 @@ pub use glob::MatchOptions;
 pub struct Glob {
     include_patterns: Vec<String>,
     exclude_patterns: Vec<Pattern>,
-    glob_match_options: MatchOptions,
 }
 
 impl Glob {
     /// Create a new [`Glob`].
     ///
-    /// Returns `None` if patterns aren't valid.
+    /// Returns `GlobError` if any of the patterns is not valid.
     pub fn new(
-        include_patterns: &[PathBuf],
-        exclude_patterns: &[PathBuf],
-        glob_match_options: MatchOptions,
-    ) -> Option<Self> {
-        let include_patterns = include_patterns
+        include_patterns: &[String],
+        exclude_patterns: &[String],
+    ) -> Result<Self, GlobError> {
+        // Validate include patterns. We can't parse the `GlobWalkers` and save them in our struct
+        // for later use because they are mutable iterators and don't implement the
+        // `std::clone::Clone` trait.
+        include_patterns
             .iter()
-            .map(|path| path.to_str().map(ToOwned::to_owned))
-            .collect::<Option<_>>()?;
+            .map(|include_pattern| -> Result<_, _> {
+                let glob =
+                    glob(include_pattern).map_err(|error| GlobError::InvalidIncludePattern {
+                        glob: include_pattern.to_owned(),
+                        error,
+                    })?;
+
+                Ok(glob)
+            })
```

Comment:
```suggestion
            .map(|include_pattern| {
                    glob(include_pattern).with_context(|| InvalidIncludePattern {
                        glob: include_pattern.to_owned()
                    })
            })
```

With the above mentioned renaming to `source` and `use` this should be possible.